### DISCUSSION TITLE
Question: Cloud Control API schema

### DISCUSSION THREAD

**Discussion by ikanni** [awslabs/aws-sdk-rust#903] [Q&A]
Question related to Cloud Control API schema:

Does the schema used by Cloud Control API is same as [CloudFormation resource provider schemas](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-type-schemas.html)

Whether the [CloudFormation resource provider schemas](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-type-schemas.html) is available in any github repo?

**Comment by jmklix** [comment]
No, they are not the same. But there are some similarities between them.

https://docs.aws.amazon.com/cloudcontrolapi/latest/userguide/what-is-cloudcontrolapi.html
> Similar to Cloud Control API, AWS CloudFormation also uses resource types to call underlying web services APIs to provision those resources when you place such a request in your account. However, CloudFormation focuses on providing resource management, by treating infrastructure as code. Using CloudFormation, you can author declarative templates that include multiple resources and their dependencies, and then provision those resources as a stack. A stack is a single unit that you then manage through AWS CloudFormation. You can also centrally manage and provision stacks across multiple AWS accounts and AWS Regions. To be managed through CloudFormation, a resource must be created as part of a stack or imported into a stack. For more information, see the [AWS CloudFormation User Guide](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/Welcome.html).

The [CDK](https://github.com/aws/aws-cdk) "synthesizes" to cloud formation. So this might be what you want to use. You can find the getting started docs [here](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html). While it doesn't support rust, you can use any of the following languages: JavaScript(TypeScript), Python, Java, .Net, or Go.

**Comment by lucas-0liveira** [comment]
The schema used by the Cloud Control API is not necessarily the same as the schemas of the CloudFormation resource provider. While both are related to cloud resource management, they may have different implementations and specifications. Therefore, it is not guaranteed that the CloudFormation resource provider schemas are available in any GitHub repository.
